### DISCUSSION TITLE
Do Range Query Item Classes in dD Spatial Searching have any other query item？

### DISCUSSION THREAD

**Discussion by WuKongBlog** [CGAL/cgal#5604] [Q&A] [answered]
My question is that I have a 3d line *l* and a distance threshold *r*, I would like to find all the points within the threshold *r* from this line *l*.  Now I‘m using the Fuzzy_iso_box class, but I find that the boundaries of this box are always parallel to the coordinate axis. So, do Range Query Item Classes have any other query item that can free rotation？

**Comment by sloriot** [comment, marked as answer]
You should be able to implement you own fuzzy query by following this [concept](https://doc.cgal.org/latest/Spatial_searching/classFuzzyQueryItem.html).

**Reply by WuKongBlog** [reply]
Thanks a lot! I'll have a try :D

**Reply by WuKongBlog** [reply]
I successfully implemented a query item that finds points inside a 3d rotate cylinder, though not looks very beautiful. I uploaded it [here](https://github.com/WuKongBlog/CGAL-Plugin/blob/main/Fuzzy_rotate_cylinder/Fuzzy_rotate_cylinder.h), hope it can help someone else.
Thank you again @sloriot 😄
